CMakeLists.txt

```diff
@@ -1,6 +1,9 @@
-cmake_minimum_required(VERSION 3.0)
+ cmake_minimum_required(VERSION 3.0...3.12)
```

Comment:
what does this exactly mean? We should try to keep the minimum required version as low as possible.

Reply:
The cmake_minimum_required function internally sets the cmake_policy to the given version,
as described in [1].
This leads to the enforcement of the given cmake_policy, even if a newer cmake version is installed/available.
The range still enforces the given minimum but internally uses a newer version if available
(in this case up to 3.12).
This allows us to use newer features while keeping the low required version.

[1] https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html